File: src/main/java/io/github/archipelagominecraft/core/loaders/legacy/LegacyForgeEntrypoint.java
//? if <=1.12.2 {
/*package io.github.archipelagominecraft.core.loaders.legacy;

import com.mojang.serialization.Codec;
import io.github.archipelagominecraft.core.ArchipelagoClientConstants;
import io.github.archipelagominecraft.core.ArchipelagoMinecraftClientCore;
import io.github.archipelagominecraft.core.compat.Logger;
import io.github.archipelagominecraft.core.forgeLike.ForgeLikeEntrypointKt;
import io.github.archipelagominecraft.core.vanilla.ArchipelagoMinecraftVanilla;
import net.minecraft.client.Minecraft;
import net.minecraftforge.fml.common.Mod;
import net.minecraftforge.fml.common.event.FMLPreInitializationEvent;
import net.minecraftforge.fml.common.event.FMLServerAboutToStartEvent;

@Mod(modid = ArchipelagoClientConstants.MOD_ID)
public class LegacyForgeEntrypoint {

    public static final Logger LOGGER =
            ArchipelagoMinecraftClientCore.LOGGER;
    @Mod.EventHandler
    public void preinit(FMLPreInitializationEvent preinit) {
        LOGGER.info("Hello, world from 1.12.2!");
        LOGGER.info("using lib: "+  Codec.STRING);
        LOGGER.info("Using minecraft class "+ Minecraft.getMinecraft().gameDir);
        ArchipelagoMinecraftClientCore.initialize();
    }

    @Mod.EventHandler
    public void init(FMLServerAboutToStartEvent preinit) {
        LOGGER.info("Legacy Forge Entrypoint initialized.");
        ForgeLikeEntrypointKt.serverAboutToStart();
    }

}

*///?}

File: mod/src/main/kotlin/io/github/archipelagominecraft/core/loaders/legacyforge/LegacyForgeEntrypoint.kt
//? if <=1.12.2 {
/*package io.github.archipelagominecraft.core.loaders.legacyforge


import cpw.mods.fml.common.event.FMLConstructionEvent
import io.github.archipelagominecraft.core.ArchipelagoClientConstants
import io.github.archipelagominecraft.core.ArchipelagoMinecraftClientCore
import io.github.archipelagominecraft.core.ArchipelagoMinecraftClientCore.LOGGER
import io.github.archipelagominecraft.core.loaders.forgeLike.ForgeLikeEntrypoint

//? if >= 1.12.2 {
typealias Mod = net.minecraftforge.fml.common.Mod
typealias EventHandler = net.minecraftforge.fml.common.Mod.EventHandler
typealias FMLPreInitializationEvent = net.minecraftforge.fml.common.event.FMLPreInitializationEvent
typealias FMLServerAboutToStartEvent = net.minecraftforge.fml.common.event.FMLServerAboutToStartEvent
//? } else {

/*typealias Mod = cpw.mods.fml.common.Mod
typealias EventHandler = cpw.mods.fml.common.Mod.EventHandler

typealias FMLPreInitializationEvent = cpw.mods.fml.common.event.FMLPreInitializationEvent
typealias FMLServerAboutToStartEvent = cpw.mods.fml.common.event.FMLServerAboutToStartEvent
*///? }

@Mod(modid = ArchipelagoClientConstants.MOD_ID)
class LegacyForgeEntrypoint {

    init {
        //? if = 1.12.2 {
        /*LOGGER.info("Hello, world from 1.12.2!")
        *///? } else if = 1.7.10 {
        /*LOGGER.info("Hello, world from 1.7.10!")
        *///? }
        ArchipelagoMinecraftClientCore.initialize()
    }
    @Suppress("unused")
    @EventHandler
    fun preinit(preinit: FMLPreInitializationEvent) {

    }

    @Suppress("unused")
    @EventHandler
    fun init(evt: FMLConstructionEvent) {
        LOGGER.info("Legacy Forge Entrypoint initialized.")
        ForgeLikeEntrypoint.fmlConstructMod()
    }


} *///?}

